src/image/config.rs

```diff
@@ -0,0 +1,304 @@
+        labels: Option<HashMap<String, String>>,
+        /// The field contains the system call signal that will be
+        /// sent to the container to exit. The signal can be a signal
+        /// name in the format SIGNAME, for instance SIGKILL or SIGRTMIN+3.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        stop_signal: Option<String>,
+    }
+);
+
+// Some fields of the image configuration are a json serialization of a
+// Go map[string]struct{} leading to the following json:
+// {
+//    "ExposedPorts": {
+//       "8080/tcp": {},
+//       "443/tcp": {},
+//    }
+// }
+// Instead we treat this as a list
+#[derive(Deserialize, Serialize)]
+struct GoMapSerde {}
+
+fn deserialize_as_vec<'de, D>(deserializer: D) -> Result<Option<Vec<String>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let opt = Option::<HashMap<String, GoMapSerde>>::deserialize(deserializer)?;
+    if let Some(data) = opt {
+        let vec: Vec<String> = data.keys().cloned().collect();
+        return Ok(Some(vec));
+    }
+
+    Ok(None)
+}
+
+fn serialize_as_map<S>(target: &Option<Vec<String>>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    match target {
+        Some(values) => {
+            let map: HashMap<_, _> = values.iter().map(|v| (v, GoMapSerde {})).collect();
+            let mut map_ser = serializer.serialize_map(Some(map.len()))?;
+            for (key, value) in map {
+                map_ser.serialize_entry(key, &value)?;
+            }
+            map_ser.end()
+        }
+        _ => unreachable!(),
+    }
+}
+
+make_pub!(
+    #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+    #[cfg_attr(
+        feature = "builder",
+        derive(derive_builder::Builder, getset::Getters),
+        builder(default, pattern = "owned", setter(into, strip_option)),
+        getset(get = "pub")
+    )]
+    /// RootFs references the layer content addresses used by the image.
+    struct RootFs {
+        /// MUST be set to layers.
+        #[serde(rename = "type")]
+        typ: String,
+        /// An array of layer content hashes (DiffIDs), in order
+        /// from first to last.
+        diff_ids: Vec<String>,
+    }
+);
+
+impl Default for RootFs {
+    fn default() -> Self {
+        Self {
+            typ: "layers".to_owned(),
+            diff_ids: Default::default(),
+        }
+    }
+}
+
+make_pub!(
+    #[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+    #[cfg_attr(
+        feature = "builder",
+        derive(derive_builder::Builder, getset::Getters),
+        builder(default, pattern = "owned", setter(into, strip_option)),
+        getset(get = "pub")
+    )]
+    /// Describes the history of a layer.
+    struct History {
+        /// A combined date and time at which the layer was created,
+        /// formatted as defined by RFC 3339, section 5.6.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        created: Option<String>,
```

Comment:
```suggestion
        #[cfg_attr(feature = "builder", getset(get = "pub"))]
        created: Option<String>,
```
also for the other fields below.